### PR TITLE
Pretty much compatible ReflectionMethod

### PR DESCRIPTION
**Pull request by asgrim**
Created new issue for `getPrototype()` cuz there's some janky logic there - see #57 plus we need to finish `ReflectionClass` first.

This resolves the `ReflectionMethod` item on #7.

### CODE REVIEW

**Comment by GeeH** [comment]
Does getInterfaces() work?

File: src/Reflection/ReflectionMethod.php [hunk 1]
```diff
@@ -50,6 +50,41 @@ public static function createFromNode(
...
+
+    /**
+     * Get the core-reflection-compatible modifier values
+     *
+     * @return int
+     */
+    public function getModifiers()
+    {
+        $val = 0;
+        $val += $this->isStatic() ? \ReflectionMethod::IS_STATIC : 0;
```
Comment (Ocramius) [inline src/Reflection/ReflectionMethod.php]:
```php
$val = ($this->isStatic() & \ReflectionMethod::IS_STATIC)
    | ($this->isPublic() & \ReflectionMethod::IS_PUBLIC)
    | ($this->isProtected() & \ReflectionMethod::IS_PROTECTED)
    | ($this->isPrivate() & \ReflectionMethod::IS_PRIVATE)
    | ($this->isAbstract() & \ReflectionMethod::IS_ABSTRACT)
    | ($this->isFinal() & \ReflectionMethod::IS_FINAL)
    | ($this->isStatic() & \ReflectionMethod::IS_STATIC)
    | ($this->isStatic() & \ReflectionMethod::IS_STATIC);
```

File: test/unit/Reflection/ReflectionMethodTest.php [hunk 2]
```diff
@@ -128,4 +131,44 @@ public function testMethodNameWithNamespace()
         $this->assertSame('', $methodInfo->getNamespaceName());
         $this->assertSame('someMethod', $methodInfo->getShortName());
     }
+
+    public function modifierProvider()
+    {
+        return [
+            ['publicMethod', 256, ['public']],
```
Comment (Ocramius) [inline test/unit/Reflection/ReflectionMethodTest.php]: Consider using internal reflection flags/constants here (in order to guarantee compat)

File: src/Reflection/ReflectionMethod.php (hunk 1 above)
Comment (Ocramius) [inline src/Reflection/ReflectionMethod.php]: Example is broken btw. `true & 1 === 1`, `true & 2 === 0`

Comment (Ocramius) [inline src/Reflection/ReflectionMethod.php]:
```php
$val = ($this->isStatic() ? 0 : \ReflectionMethod::IS_STATIC)
    | ($this->isPublic() ? 0 : \ReflectionMethod::IS_PUBLIC)
    | ($this->isProtected() ? 0 : \ReflectionMethod::IS_PROTECTED)
    | ($this->isPrivate() ? 0 : \ReflectionMethod::IS_PRIVATE)
    | ($this->isAbstract() ? 0 : \ReflectionMethod::IS_ABSTRACT)
    | ($this->isFinal() ? 0 : \ReflectionMethod::IS_FINAL)
    | ($this->isStatic() ? 0 : \ReflectionMethod::IS_STATIC)
    | ($this->isStatic() ? 0 : \ReflectionMethod::IS_STATIC);
```

So yeah, ternaries -.-

**Comment by asgrim** [comment]
@GeeH `getInterfaces()` does not yet work - I thought you were implementing it? ;)

File: src/Reflection/ReflectionMethod.php (hunk 1 above)
Comment (asgrim) [inline src/Reflection/ReflectionMethod.php]: That.. also doesn't work.

File: test/unit/Reflection/ReflectionMethodTest.php (hunk 2 above)
Comment (asgrim) [inline test/unit/Reflection/ReflectionMethodTest.php]: Good idea - I've updated, ready for review again please Mr. @Ocramius

File: src/Reflection/ReflectionMethod.php (hunk 1 above)
Comment (Ocramius) [inline src/Reflection/ReflectionMethod.php]: Alright, guess I'll just be ok with the current approach.

**Comment by Ocramius** [comment]
:+1:

**Pull request merged**